File: Caesar_Cipher.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

const string DS = "0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^-'abcdefghijklmnopqrstuvwxyz{|}~ ";

string encrypt(string plaintext, int key) {
    string encryptedText = "";
    for (char& c : plaintext) {
        if (DS.find(c) != string::npos) {
            int index = (DS.find(c) + key) % DS.size();
            encryptedText += DS[index];
        } else {
            encryptedText += c;
        }
    }
    return encryptedText;
}

string decrypt(string encryptedText, int key) {
    string plaintext = "";
    for (char& c : encryptedText) {
        if (DS.find(c) != string::npos) {
            int index = (DS.find(c) - key + DS.size()) % DS.size();
            plaintext += DS[index];
        } else {
            plaintext += c;
        }
    }
    return plaintext;
}

int main() {
    srand(time(nullptr));

    cout << "\n\t\t Caesar Cipher Encrypt/Decrypt Application\n\n ";
    cout << "1. Encrypt a message\n";
    cout << "2. Decrypt a message\n";
    cout << ">>";
    string menuChoice;
    cin >> menuChoice;

    if (menuChoice == "1") {
        string plaintext;
        cout << "\nEnter the message you want to encrypt : ";
        getline(cin >> ws, plaintext);
        while (plaintext.length() < 15) {
            cout << "The message should be 15 characters minimum, Enter the message again: ";
            getline(cin >> ws, plaintext);
        }

        int key = rand() % 75 + 1;
        string encryptedText = encrypt(plaintext, key);

        ofstream outfile("encryption.txt");
        outfile << encryptedText;
        outfile.close();

        cout << "\nOriginal Message (Plaintext): " << plaintext << endl;
        cout << "Key: " << key << endl;
        cout << "Encryption result: " << encryptedText << endl;
        cout << "\nEncryption result saved in a file\n";
    } else if (menuChoice == "2") {
        string encryptedText;
        cout << "\nEnter the encrypted text you want to decrypt : ";
        getline(cin >> ws, encryptedText);
        int key;
        cout << "Enter the key : ";
        cin >> key;

        string decrypted_plaintext = decrypt(encryptedText, key);

        ofstream outfile("decryption.txt");
        outfile << decrypted_plaintext;
        outfile.close();

        cout << "Decrypted result: " << decrypted_plaintext << endl;
        cout << "\nDecryption result saved in a file\n";
    } else {
        cout << "Choose 1 or 2\n";
    }

    return 0;
}

File: Caesar_Cipher.py
# libs
import random

# The characters that can be used in the process, the user can use only use this set of chars 
DS = "0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^-'abcdefghijklmnopqrstuvwxyz{|}~"

# Encryption function
def encrypt(plaintext, key):
    encryptedText = ""
    for char in plaintext:
        if char in DS:
            index = (DS.index(char) + key) % len(DS)
            encryptedText += DS[index]
        else:
            encryptedText += char
    return encryptedText

# Decryption function
def decrypt(encryptedText, key):
    plaintext = ""
    for char in encryptedText:
        if char in DS:
            index = (DS.index(char) - int(key)) % len(DS)
            plaintext += DS[index]
        else:
            plaintext += char
    return plaintext

# App Menu
print("\n\t\t Caesar Cipher Encrypt/Decrypt Application\n\n ")
print("1. Encrypt a message\n")
print("2. Decrypt a message\n")
menuChoice = input(">>")


if(menuChoice == "1"):

    plaintext = input("\nEnter the message you want to encrypt : ")
    while len(plaintext) < 15:
        plaintext = input("The message should be 15 characters minimum, Enter the message again:  ")
    # Key Generation
    key = random.randint(1, 75)
    # Encrypt with key
    encryptedText = encrypt(plaintext, key)
    with open("encryption", "w") as f:
        f.write(encryptedText)
    print(f"\nOriginal Message (Plaintext): {plaintext}")
    print(f"Key: {key}")
    print(f"Encryption result: {encryptedText}")
    print("\nEncryption result saved in a file\n")

elif(menuChoice == "2"):
    encryptedText = input("\nEnter the encrypted text you want to decrypt : ")
    key = input("Enter the key : ")
    # Decrypt with key
    decrypted_plaintext = decrypt(encryptedText, key)
    # Saving the result into a file
    with open("decryption.txt", "w") as f:
        f.write(decrypted_plaintext)
    print(f"Decrypted result: {decrypted_plaintext}")
    print("\nDecryption result saved in a file\n")

else:
    print("Choose 1 or 2")

"""
Example #1

Original Message (Plaintext): Ahmad Raed Jawabreh
Key: 8
Encryption result: Ipuil Ziml Ri0ijzmp

Exapme #2
Original Message (Plaintext): i am ahmad jawabreh
Key: 26
Encryption result: 4 {8 {38{~ 5{B{|=03


"""